x/csr/keeper/integration_test.go

```diff
@@ -68,25 +72,62 @@ var _ = Describe("CSR Distribution : ", Ordered, func() {
 		s.Commit()
 
 		// deploy the turnstile
-		turnstileAddress = deployContract(deployerKey, turnstileCode)
+		turnstileAddress, _ = s.app.CSRKeeper.GetTurnstile(s.ctx)
 		s.Commit()
 
+		// Deploy a test smart contract
+		contractAddress = deployContract(deployerKey, contractCode)
+		s.Commit()
 	})
 
 	Context("testing init", func() {
-		It("It should exist", func() {
```

Comment:
It is possible to separate this Context into its own unit-test? That way, we can have a clear separation of responsibility it writing the integration tests / editing existing tests.

Reply:
yea i am writing the unit tests for the event handlers independent of the integration tests.